dlc-messages/src/ser_impls.rs

```diff
@@ -537,6 +538,23 @@ pub fn read_i32<R: ::std::io::Read>(reader: &mut R) -> Result<i32, DecodeError>
     Ok(i32::from_be_bytes(v))
 }
 
+/// Writes an `i64` value to the given writer.
+pub fn write_i64<W: Writer>(i: &i64, writer: &mut W) -> Result<(), ::std::io::Error> {
+    let i = i.to_be_bytes();
+    let i1: [u8; 4] = i[0..4].try_into().unwrap();
+    let i2: [u8; 4] = i[4..8].try_into().unwrap();
+    i1.write(writer)?;
+    i2.write(writer)
+}
+
+/// Reads an `i64` value from the given reader.
+pub fn read_i64<R: ::std::io::Read>(reader: &mut R) -> Result<i64, DecodeError> {
+    let v1: [u8; 4] = Readable::read(reader)?;
+    let v2: [u8; 4] = Readable::read(reader)?;
+    let v = [v1, v2].concat();
+    Ok(i64::from_be_bytes(v.try_into().unwrap()))
+}
```

Comment:
What about just writing/reading byte per byte? Did some quick benchmark and seems much faster for the reading (somehow the writing benchmark doesn't work so couldn't compare).
```suggestion
/// Writes an `i64` value to the given writer.
pub fn write_i64<W: Writer>(i: &i64, writer: &mut W) -> Result<(), ::std::io::Error> {
    let i = i.to_be_bytes();
    for b in i {
        b.write(writer)?;
    }
    Ok(())
}

/// Reads an `i64` value from the given reader.
pub fn read_i64<R: ::std::io::Read>(reader: &mut R) -> Result<i64, DecodeError> {
    let mut v = [0u8; 8];
    for i in 0..8 {
        v[i] = Readable::read(reader)?;
    }
    Ok(i64::from_be_bytes(v))
}
```

Reply:
that crossed my mind, and after seeing it implemented, i like it a lot more! lets go with it

Reply:
Wooops made clippy unhappy sorry...